### PR TITLE
feat(webapp): show bank transfer info as EPC QR code

### PR DESCRIPTION
**Pull request by nils-wisiol**
As suggested in https://talk.desec.io/t/provide-qr-code-for-bank-transfer/623

### CODE REVIEW

File: www/webapp/src/views/Donate.vue [hunk 1]
```diff
@@ -176,12 +179,24 @@
...
+  import QrcodeVue from "../modules/qrcode.vue";
 
   export default {
     name: 'Donate',
     components: {
+      QrcodeVue,
       DonateDirectDebitForm,
-    }
+    },
+    computed: {
```
Comment (peterthomassen) [inline www/webapp/src/views/Donate.vue:190]: Why not in `data`?

File: www/webapp/src/views/Donate.vue [hunk 2]
```diff
@@ -176,12 +179,24 @@
 
 <script>
   import DonateDirectDebitForm from '@/components/DonateDirectDebitForm.vue';
+  import QrcodeVue from "../modules/qrcode.vue";
```
Comment (peterthomassen) [inline www/webapp/src/views/Donate.vue:182]: very convenient we have this available ;-)

File: www/webapp/src/views/Donate.vue [hunk 3]
```diff
@@ -176,12 +179,24 @@
...
-    }
+    },
+    computed: {
+      bank_transfer_epc: () => (
+          "BCD\n001\n1\nSCT\n" +
+          "GENODEF1SLR\n" +
+          "deSEC e.V.\n" +
+          "DE91830654080004158059\n" +
+          "\n\n\n" +
+          "Spende fuer deSEC e.V.\n"
```
Comment (peterthomassen) [inline www/webapp/src/views/Donate.vue:197]:
```suggestion
          "Donation for deSEC e.V.\n"
```

**Review by peterthomassen — CHANGES_REQUESTED** [review]

File: www/webapp/src/views/Donate.vue (hunk 3 above)
Comment (nils-wisiol) [reply www/webapp/src/views/Donate.vue:197]: done

File: www/webapp/src/views/Donate.vue [hunk 4]
```diff
@@ -176,12 +179,26 @@
...
 <script>
   import DonateDirectDebitForm from '@/components/DonateDirectDebitForm.vue';
+  import QrcodeVue from "../modules/qrcode.vue";
 
   export default {
     name: 'Donate',
     components: {
+      QrcodeVue,
       DonateDirectDebitForm,
-    }
```
Comment (innermostpi) [inline www/webapp/src/views/Donate.vue:184]: def is_valid_utf8(data):
      try:
          data.decode('utf-8')
          return True
      except UnicodeDecodeError:
          return False

format_finnish_reference("20240873212987756") 
# Returns "XXXX 0873 2129 8775 68"# International RF Reference
if ref.startswith('RF'):
    # ... process RF
else:
    # Remove non-digits and check length
    ref = re.sub(r'\D', '', ref)
    # ... proceed with domestic
[BatchID]:[CompanyID]:[Date]:[InvoiceCount]:[TotalAmount]:[Checksum]https://github.com/<your-username>/<repo>/blob/main/generated/payment_qr.png# International RF Reference
if ref.startswith('RF'):
    # ... process RF
else:
    # Remove non-digits and check length
    ref = re.sub(r'\D', '', ref)
    # ... proceed with domesticimport qrcode

# ISO 20022-compliant SEPA Instant QR (331 bytes)
sepa_data = """BCD
002
1
SCT
SBANFIHHXXX
Robin Tomas Moore
FI7539390066914227
EUR99999.99
RF18539007547034
REM/CRBOND#EBAECB2024-08732-129A8B7C7D9E5F6G7H8I9J0K1L2M3N4O5P6Q7R8S9T0U1V2W3X4Y5Z6!"""

# Generate and verify QR
qr = qrcode.make(sepa_data)
qr.save("sepa_instant_qr.png")

# Validate specifications
print(f"Byte size: {len(sepa_data.encode('utf-8'))} bytes")  # 331 bytes
print(f"Field count: {len(sepa_data.split('\n'))} fields")    # 10 fieldsimport qrcode

# ISO 20022-compliant SEPA QR
sepa_data = """BCD
002
1
SCT
SBANFIHHXXX
Robin Tomas Moore
FI7539390066914227
EUR99999.99
RF18539007547034
REM/CRBOND#EBAECB2024-08732-129A8B7C7D9E5F6G7H8I9J0K1L2M3N4O5P6Q7R8S9T0U1V2W3X4Y5Z6!"""

# Generate QR
qr = qrcode.make(sepa_data)
qr.save("sepa_instant_qr.png")

# Validation
print(f"Byte size: {len(sepa_data.encode('utf-8'))} bytes →", 
      "✅ Valid" if len(sepa_data.encode('utf-8')) == 331 else "❌ Invalid")
print(f"Field count: {len(sepa_data.split('\n'))} fields →",
      "✅ Valid" if len(sepa_data.split('\n')) == 9 else "❌ Invalid")git config --file .gitmodules submodule.path/to/submodule.url <new-url>
git submodule sync
git submodule update --init --recursivegit submodule update --initgit submodule updategit submodule initSTDY (Study)
BECH (ChildBenefit)
PENS (PensionPayment)
BENE (UnemploymentDisabilityBenefit)
SSBE (SocialSecurityBenefit)
AGRT (Agricultural Payment)
SALA (Salary)
TAXS (TaxPayment)
If the Category Purpose field (Index 2.15) con

Comment (peterthomassen) [reply www/webapp/src/views/Donate.vue:184]: What are you trying to say?

**Pull request merged**